### PR TITLE
Feature/custom message

### PR DESCRIPTION
**Pull request by Zemux1613**
#15

### CODE REVIEW

**Comment by OliverSchlueter** [comment]
I think your branch was outdated

**Comment by Zemux1613** [comment]
now

File: src/main/java/de/oliver/fancynpcs/utils/EntityTypes.java [hunk 1]
```diff
@@ -38,23 +39,25 @@ public class EntityTypes {
...
 
-                if(excludedTypes.contains(type)){
+                if (excludedTypes.contains(type)) {
                     continue;
                 }
 
                 TYPES.put(type.toShortString(), type);
-            } catch (Exception e){ }
+            } catch (Exception ex) {
+                ex.printStackTrace();
```
Comment (OliverSchlueter) [inline src/main/java/de/oliver/fancynpcs/utils/EntityTypes.java:59]: The `ex.printStackTrace();` needs to be removed, it spams the console

File: src/main/java/de/oliver/fancynpcs/commands/NpcCMD.java [hunk 2]
```diff
@@ -168,56 +170,83 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
...
+                if (!npcModifyEvent.isCancelled()) {
                     npc.moveForAll(p.getLocation());
                     MessageHelper.success(sender, "Moved NPC to your location");
                 } else {
                     MessageHelper.error(sender, "Modification has been cancelled");
                 }
             }
 
+            case "message" -> {
+                if (args.length >= 3) {
```
Comment (OliverSchlueter) [inline src/main/java/de/oliver/fancynpcs/commands/NpcCMD.java:190]: What if the message is longer than one word?

File: src/main/java/de/oliver/fancynpcs/commands/NpcCMD.java [hunk 3]
```diff
@@ -168,56 +170,83 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
...
+
+                NpcModifyEvent npcModifyEvent = new NpcModifyEvent(npc, NpcModifyEvent.NpcModification.CUSTOM_MESSAGE, p);
+                npcModifyEvent.callEvent();
+
+                if (!npcModifyEvent.isCancelled()) {
+                    StringBuilder message = new StringBuilder();
+                    for (int i = 2; i < args.length; i++) {
+                        message.append(args[i]);
+                    }
+                    npc.setOnClick(player -> player.sendMessage(Component.text(message.toString())));
```
Comment (OliverSchlueter) [inline src/main/java/de/oliver/fancynpcs/commands/NpcCMD.java:209]: This should be in Npc#updateMessage

**Review by OliverSchlueter — CHANGES_REQUESTED** [review]

**Review by OliverSchlueter — APPROVED** [review]

**Pull request merged**